public/app/features/dashboard-scene/scene/layout-rows/RowsLayoutManager.tsx

```diff
@@ -16,9 +17,29 @@ interface RowsLayoutManagerState extends SceneObjectState {
 export class RowsLayoutManager extends SceneObjectBase<RowsLayoutManagerState> implements DashboardLayoutManager {
   public isDashboardLayoutManager: true = true;
 
+  private _context: ElementSelectionContextState | undefined;
+
   public editModeChanged(isEditing: boolean): void {}
 
-  public addPanel(vizPanel: VizPanel): void {}
+  public addPanel(vizPanel: VizPanel): void {
+    // Try to add new panels to all selected rows
+    if (this._context?.enabled && this._context?.selected.length > 0) {
+      return this.state.rows
+        .filter((r) => this._context?.selected.some((s) => s.id === r.state.key))
+        .forEach((r) => {
+          r.onAddPanel(vizPanel.clone());
+        });
```

Comment:
don't think we need this, we can just get the editPane here (from the DashboardScene). It has the selectedObject so easy to check what is selected

Reply:
Thanks! Just updated!